src/legacy/ui/public/index_patterns/_index_pattern.js

```diff
@@ -18,456 +18,418 @@
+import { mappingSetupService } from 'ui/utils/mapping_setup';
+import { toastNotifications } from 'ui/notify';
+import { findObjectByTitle } from 'ui/saved_objects';
 
+import { IndexPatternMissingIndices } from './errors';
 import { getComputedFields } from './_get_computed_fields';
-import { formatHit } from './_format_hit';
-import { IndexPatternsGetProvider } from './_get';
+import { getRoutes } from './get_routes';
+import { formatHitProvider } from './_format_hit';
 import { FieldList } from './_field_list';
-import { IndexPatternsFlattenHitProvider } from './_flatten_hit';
-import { IndexPatternsPatternCacheProvider } from './_pattern_cache';
-import { FieldsFetcherProvider } from './fields_fetcher_provider';
-import { SavedObjectsClientProvider, findObjectByTitle } from '../saved_objects';
+import { flattenHitWrapper } from './_flatten_hit';
+import { getConfig } from './config';
+
 import { i18n } from '@kbn/i18n';
 
-export function getRoutes() {
-  return {
-    edit: '/management/kibana/index_patterns/{{id}}',
-    addField: '/management/kibana/index_patterns/{{id}}/create-field',
-    indexedFields: '/management/kibana/index_patterns/{{id}}?_a=(tab:indexedFields)',
-    scriptedFields: '/management/kibana/index_patterns/{{id}}?_a=(tab:scriptedFields)',
-    sourceFilters: '/management/kibana/index_patterns/{{id}}?_a=(tab:sourceFilters)'
-  };
-}
 
 const MAX_ATTEMPTS_TO_RESOLVE_CONFLICTS = 3;
 
-export function IndexPatternProvider(Private, config, Promise) {
-  const getConfig = (...args) => config.get(...args);
-  const getIds = Private(IndexPatternsGetProvider)('id');
-  const fieldsFetcher = Private(FieldsFetcherProvider);
-  const mappingSetup = Private(UtilsMappingSetupProvider);
-  const flattenHit = Private(IndexPatternsFlattenHitProvider);
-  const patternCache = Private(IndexPatternsPatternCacheProvider);
-  const savedObjectsClient = Private(SavedObjectsClientProvider);
-  const fieldformats = fieldFormats;
-
-  const type = 'index-pattern';
-  const configWatchers = new WeakMap();
-
-  const mapping = mappingSetup.expandShorthand({
-    title: 'text',
-    timeFieldName: 'keyword',
-    intervalName: 'keyword',
-    fields: 'json',
-    sourceFilters: 'json',
-    fieldFormatMap: {
-      type: 'text',
-      _serialize(map = {}) {
-        const serialized = _.transform(map, serializeFieldFormatMap);
-        return _.isEmpty(serialized) ? undefined : angular.toJson(serialized);
-      },
-      _deserialize(map = '{}') {
-        return _.mapValues(angular.fromJson(map), deserializeFieldFormatMap);
-      }
-    },
-    type: 'keyword',
-    typeMeta: 'json',
+
+
+const type = 'index-pattern';
+
+function serializeFieldFormatMap(flat, format, field) {
+  if (format) {
+    flat[field] = format;
+  }
+}
+
+function deserializeFieldFormatMap(mapping) {
+  const FieldFormat = fieldFormats.byId[mapping.id];
+  return FieldFormat && new FieldFormat(mapping.params, getConfig);
```

Comment:
I'm struggling to find the type returned by `fieldFormats.byId` but if it's https://github.com/elastic/kibana/blob/master/src/legacy/ui/field_formats/field_format.js#L23 then it seems like the `FieldFormat` constructor only takes one argument and we can get rid of `getConfig` parameter.

Reply:
nice catch! 
i also move all methods into the class